### ISSUE TITLE
FlameBlocListenable error when revisiting a Route

### ISSUE THREAD

**Issue by pastordougdev** [flame-engine/flame#2095]
# Current bug behaviour
After visiting a Route containing a component that uses FlameBlocListenable and popping that route, if the same route is visited again, a LateInitializationError occurs when the onMount method of the mixin sets the _subscription variable.

See below for a link to a Github repo with a small Demo app to reproduce this issue.

# Expected behaviour
Expecting that the component(s) on the route can access the Bloc normally.

# Steps to reproduce
Push any route having a component using FlameBlocListenable mixin.
Pop that route.
Push the previously visited route again.
Will receive a LateInitializationError on the _subscription field in the onMount method.

# Flutter doctor output

```
Output of: flutter doctor -v
pastordoug@Douglass-MBP bug_demo % flutter doctor -v
[✓] Flutter (Channel stable, 3.3.2, on macOS 11.6 20G165 darwin-x64, locale en-US)
    • Flutter version 3.3.2 on channel stable at /Users/pastordoug/Flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e3c29ec00c (5 weeks ago), 2022-09-14 08:46:55 -0500
    • Engine revision a4ff2c53d8
    • Dart version 2.18.1
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pastordoug/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2021.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 66.0.3
    • Dart plugin version 213.7371

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] VS Code (version 1.55.2)
    • VS Code at /Users/pastordoug/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.6 20G165 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 106.0.5249.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

# More environment information

# Log information

```
======== Exception caught by scheduler library =====================================================
The following LateError was thrown during a scheduler callback:
LateInitializationError: Field '_subscription' has already been initialized.

When the exception was thrown, this was the stack: 
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 266:49      throw_
packages/flame_bloc/src/flame_bloc_listenable.dart 11:36                          set [_subscription]
packages/flame_bloc/src/flame_bloc_listenable.dart 53:5                           onMount
packages/flame/src/components/core/component.dart 784:5                           [_mount]
packages/flame/src/components/core/component.dart 792:25                          <fn>
dart-sdk/lib/collection/iterable.dart 45:35                                       forEach
packages/flame/src/components/core/component.dart 791:7                           [_mount]
packages/flame/src/components/core/component.dart 983:14                          [_processChildrenQueue]
packages/flame/src/components/core/component.dart 973:5                           processQueues
packages/flame/src/components/core/component.dart 460:24                          updateTree
packages/flame/src/components/core/component.dart 462:32                          <fn>
dart-sdk/lib/collection/iterable.dart 45:35                                       forEach
packages/flame/src/components/core/component.dart 462:16                          updateTree
packages/flame/src/game/flame_game.dart 82:30                                     <fn>
dart-sdk/lib/collection/iterable.dart 45:35                                       forEach
packages/flame/src/game/flame_game.dart 82:14                                     updateTree
packages/flame/src/game/flame_game.dart 70:7                                      update
packages/flame/src/game/game_render_box.dart 53:10                                gameLoopCallback
packages/flame/src/game/game_loop.dart 46:13                                      [_tick]
packages/flutter/src/scheduler/ticker.dart 249:12                                 [_tick]
packages/flutter/src/scheduler/binding.dart 1175:15                               [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1079:11                               <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/linked_hash_map.dart 21:13          forEach
packages/flutter/src/scheduler/binding.dart 1077:16                               handleBeginFrame
packages/flutter/src/scheduler/binding.dart 994:5                                 [_handleBeginFrame]
lib/_engine/engine/platform_dispatcher.dart 1183:13                               invoke1
lib/_engine/engine/platform_dispatcher.dart 198:5                                 invokeOnBeginFrame
lib/_engine/engine/initialization.dart 186:45                                     <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 334:14  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 339:39  dcall
```

# More information

## Demo
I put together a small Demo application to recreate this issue.  More details are in the readme in this repo:
[Demo App] (https://github.com/pastordougdev/bug_demo_1)
## Working on PR
It would my pleasure to work on a PR for this.

**Comment by spydon** [comment]
@erickzanardo @felangel

**Comment by spydon** [comment]
> It would my pleasure to work on a PR for this.

Ah, I missed this, I'll assign you @pastordougdev! :)

**Comment by erickzanardo** [comment]
Good catch!

From an initial look at the code, I think we need to drop the `final` from the subscription declaration: https://github.com/flame-engine/flame/blob/main/packages/flame_bloc/lib/src/flame_bloc_listenable.dart#L11

Since the same component can be removed and mounted again. I am unsure if that is all, but definitely where I would start. @pastordougdev lmk if you need any help with the PR

**Comment by pastordougdev** [comment]
In my current project, I cloned the mixin and dropped the `final` from the declaration and the component remounts without error.  Since the subscription is being canceled in onRemove, it seemed safe to me to be able to reassign subscription in the subsequent onMount.

@erickzanardo thanks for the offer of assistance.  Will be my first PR so I will reach out for any questions!  Thanks!

**Comment by erickzanardo** [comment]
Fixed on #2098
